### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by jasonwbarnett**
Not sure if these ever worked, but I couldn't get them to pass and upon close investigation the tests themselves weren't accurate.

### CODE REVIEW

File: pkg/artifactory/resource_artifactory_local_repository_test.go [hunk 1]
```diff
@@ -56,7 +56,6 @@ resource "artifactory_local_repository" "terraform-local-test-repo-full" {
 	calculate_yum_metadata          = false
 	yum_root_depth                  = 0
 	docker_api_version              = "V2"
-	force_nuget_authentication		= true
```
Comment (jasonwbarnett) [inline pkg/artifactory/resource_artifactory_local_repository_test.go:59]: You can't enable `forceNugetAuthentication` on a npm repository. I'm not sure why this ever worked or why this was added.

File: pkg/artifactory/resource_artifactory_local_repository_test.go [hunk 2]
```diff
@@ -90,7 +89,6 @@ func TestAccLocalRepository_full(t *testing.T) {
 					resource.TestCheckResourceAttr("artifactory_local_repository.terraform-local-test-repo-full", "calculate_yum_metadata", "false"),
 					resource.TestCheckResourceAttr("artifactory_local_repository.terraform-local-test-repo-full", "yum_root_depth", "0"),
 					resource.TestCheckResourceAttr("artifactory_local_repository.terraform-local-test-repo-full", "docker_api_version", "V2"),
-					resource.TestCheckResourceAttr("artifactory_local_repository.terraform-local-test-repo-full", "force_nuget_authentication", "true"),
```
Comment (jasonwbarnett) [inline pkg/artifactory/resource_artifactory_local_repository_test.go:93]: You can't enable `forceNugetAuthentication` on a npm repository. I'm not sure how this ever worked or why this was added.

**Comment by chb0github** [comment]
Let me give you some background on this repo; that should help guide your thinking. Although your PR is simple, I am not sure I can approve it (I need to think on it).

We inherited this codebase from atlassian a while back. The current "design" is basically that you can create "generic repos+". So, what I mean is that, you can create a generic repo only, but those generic repos also have some misc optional flags that only apply to certain repos, and inconsitently so. The right approach is to have a different resource type per repository so terraform can enforce and allow these features. 

It's not a 'no', since you're only modifying tests. But, there are a lot of other tests and I am not sure why you chose these to fix when you have no code changes and no other fixes

**Comment by jasonwbarnett** [comment]
@chb0github to give you some context on why I did this in the first place. I had a desire to make some significant changes to the underlying code, but I wanted to ensure I could get the current test suite passing before doing so. I started down the path of verifying the current test suite works and I found that it doesn't. I looked at the tests and in my evaluation it made the most sense to make the changes I did in order to get them to pass because in their current state that don't provide as much value as if they would if they passed.

There are others that aren't passing either, but were more complex and I couldn't make an honest evaluation as to which changes would be appropriate so I'm holding off on others.

Are you able to get the current test suite passing without any issues?

**Comment by chb0github** [comment]
The current test suite does not pass and it's been something I have been banging the drum to fix. 

In a nutshell, a big reason the other tests don't work is that there is some quasi-client code written as part of the tests and it doesn't conform with changes that have occured since. 

I can supply you with a test enterprise instance to help you along and review it with you, etc.

**Comment by jasonwbarnett** [comment]
@chb0github That all makes sense. It would be great if you could supply me with a test enterprise license to help with the development of this project.

**Comment by chb0github** [comment]
I can't give you a test license, but I can give you access to an enterprise instance; which should be good enough

**Comment by chb0github** [comment]
Ok, I have a test instance - please PM in gitter

**Comment by chb0github** [comment]
Where are we on this? Did you get testing working? Working tests and your other requests have bubbled much closer to the top of getting done

**Comment by jasonwbarnett** [comment]
@chb0github This MR gets you closer to a passing test suite. I haven't made the time to invest in fixing the remaining failing tests.

File: pkg/artifactory/resource_artifactory_local_repository_test.go (hunk 2 above)
Comment (chb0github) [reply pkg/artifactory/resource_artifactory_local_repository_test.go:93]: This is what I meant by "repository +"

**Review by chb0github — APPROVED** [review]
Accepting

**Pull request merged**